lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -658,7 +658,10 @@ sub _create_clone {
     # not be created, somebody else was faster at cloning)
     my $orig_id       = $self->id;
     my $affected_rows = $rset->search({id => $orig_id, clone_id => undef})->update({clone_id => $new_job->id});
-    die "Job $orig_id has already been cloned as " . $self->clone_id . "\n" unless $affected_rows == 1;
+    unless ($affected_rows == 1) {
+        $self->discard_changes;
+        die "Job $orig_id has already been cloned as " . $self->clone_id . "\n";
+    }
```

Comment:
Would that be equivalent to

```perl
die "Job $orig_id has already been cloned as " . $rset->search($orig_id)->clone_id . "\n" unless $affected_rows == 1;
```

Reply:
No, but something similar. I pushed a change.